tests/test_v2.py

```diff
@@ -0,0 +1,346 @@
+    dtype, value = dtype_value
+    with config.set(
+        {
+            "array.v2_default_filters": {
+                "string": [{"id": "vlen-utf8"}],
+                "bytes": [{"id": "vlen-bytes"}],
+            },
+        }
+    ):
+        expected = np.full((3,), value, dtype=dtype)
+        a = zarr.create(
+            store=tmp_path,
+            shape=(3,),
+            zarr_format=2,
+            dtype=dtype,
+        )
+        a[:] = expected
+        data = a[:]
+        np.testing.assert_equal(data, expected)
+
+
+@pytest.mark.parametrize("dtype", [str, "str"])
+async def test_create_dtype_str(dtype: Any) -> None:
+    arr = zarr.create(shape=3, dtype=dtype, zarr_format=2)
+    assert arr.dtype.kind == "O"
+    assert arr.metadata.to_dict()["dtype"] == "|O"
+    assert arr.metadata.filters == (numcodecs.vlen.VLenBytes(),)
+    arr[:] = [b"a", b"bb", b"ccc"]
+    result = arr[:]
+    np.testing.assert_array_equal(
+        result, np.array([b"a", b"bb", b"ccc"], dtype="object")
+    )
+
+
+@pytest.mark.parametrize(
+    "filters", [[], [numcodecs.Delta(dtype="<i4")], [numcodecs.Zlib(level=2)]]
+)
+@pytest.mark.parametrize("order", ["C", "F"])
+def test_v2_filters_codecs(
+    filters: Any, order: Literal["C", "F"], tmp_path: Path
+) -> None:
+    array_fixture = [42]
+    with config.set({"array.order": order}):
+        arr = zarr.create(
+            store=tmp_path, shape=1, dtype="<i4", zarr_format=2, filters=filters
+        )
+    arr[:] = array_fixture
+    result = arr[:]
+    np.testing.assert_array_equal(result, array_fixture)
+
+
+@pytest.mark.filterwarnings("ignore")
+@pytest.mark.parametrize("store", ["memory"], indirect=True)
+def test_create_array_defaults(store: Store):
+    """
+    Test that passing compressor=None results in no compressor. Also test that the default value of the compressor
+    parameter does produce a compressor.
+    """
+    g = zarr.open(store, mode="w", zarr_format=2)
+    arr = g.create_array("one", dtype="i8", shape=(1,), chunks=(1,), compressor=None)
+    assert arr._async_array.compressor is None
+    assert not (arr.filters)
+    arr = g.create_array("two", dtype="i8", shape=(1,), chunks=(1,))
+    assert arr._async_array.compressor is not None
+    assert not (arr.filters)
+    arr = g.create_array(
+        "three", dtype="i8", shape=(1,), chunks=(1,), compressor=Zstd()
+    )
+    assert arr._async_array.compressor is not None
+    assert not (arr.filters)
+    with pytest.raises(ValueError):  # noqa: PT011
+        g.create_array(
+            "four",
+            dtype="i8",
+            shape=(1,),
+            chunks=(1,),
+            compressor=None,
+            compressors=None,
+        )
+
+
+@pytest.mark.parametrize(
+    "array_order", ["C", pytest.param("F", marks=[pytest.mark.xfail])]
+)
+@pytest.mark.parametrize("data_order", ["C", "F"])
+@pytest.mark.parametrize(
+    "memory_order", ["C", pytest.param("F", marks=[pytest.mark.xfail])]
+)
```

Comment:
Should we fall back to python for F-arrays or just fail?

Reply:
We could handle `F` array order easily if `zarr_python` gave us all of the array metadata when constructing the codec pipeline

https://github.com/ilan-gold/zarrs-python/blob/26ee5166ca4473c5f96a7cdf432779cd663ea586/src/metadata_v2.rs#L36-L39